ibm/resource_ibm_cis_certificate_upload.go

```diff
@@ -0,0 +1,354 @@
+package ibm
+
+import (
+	"log"
+	"time"
+
+	"github.com/IBM/go-sdk-core/v3/core"
+	cissslv1 "github.com/IBM/networking-go-sdk/sslcertificateapiv1"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	ibmCISCertificateUpload             = "ibm_cis_certificate_upload"
+	cisCertificateUploadCertificate     = "certificate"
+	cisCertificateUploadPrivateKey      = "private_key"
+	cisCertificateUploadBundleMethod    = "bundle_method"
+	cisCertificateUploadGeoRestrictions = "geo_restrictions"
+	cisCertificateUploadCustomCertID    = "custom_cert_id"
+	cisCertificateUploadStatus          = "status"
+	cisCertificateUploadPriority        = "priority"
+	cisCertificateUploadHosts           = "hosts"
+	cisCertificateUploadIssuer          = "issuer"
+	cisCertificateUploadSignature       = "signature"
+	cisCertificateUploadUploadedOn      = "uploaded_on"
+	cisCertificateUploadModifiedOn      = "modified_on"
+	cisCertificateUploadExpiresOn       = "expires_on"
+	cisCertificateUploadUbiquitous      = "ubiquitous"
+	cisCertificateUploadDeletePending   = "deleting"
+	cisCertificateUploadDeleted         = "deleted"
+)
+
+func resourceIBMCISCertificateUpload() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceCISCertificateUploadCreate,
+		Read:     resourceCISCertificateUploadRead,
+		Update:   resourceCISCertificateUploadUpdate,
+		Delete:   resourceCISCertificateUploadDelete,
+		Exists:   resourceCISCertificateUploadExists,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisCertificateUploadCustomCertID: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			cisCertificateUploadCertificate: {
+				Type:        schema.TypeString,
+				Description: "Certificate key",
+				Required:    true,
+				Sensitive:   true,
+			},
+			cisCertificateUploadPrivateKey: {
+				Type:        schema.TypeString,
+				Description: "Certificate private key",
+				Required:    true,
+				Sensitive:   true,
+			},
+			cisCertificateUploadBundleMethod: {
+				Type:        schema.TypeString,
+				Description: "Certificate bundle method",
+				Optional:    true,
+				Default:     cisCertificateUploadUbiquitous,
+				ValidateFunc: InvokeValidator(
+					ibmCISCertificateUpload,
+					cisCertificateUploadBundleMethod),
+			},
+			cisCertificateUploadHosts: {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "hosts which the certificate uploaded to",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			cisPageRulePriority: {
```

Comment:
This is not used any where..
Please use in payload.. and set in read method.. both in resource and datasource

Reply:
changed to `cisCertificateUploadPriority`